src/postgres/commands/PostgresQueryWizard/FunctionQuerySteps/FunctionQueryReturnTypeStep.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { QuickPickItem } from "vscode";
+import { AzureWizardPromptStep, IWizardOptions } from "vscode-azureextensionui";
+import { ext } from "../../../../extensionVariables";
+import { localize } from "../../../../utils/localize";
+import { FunctionQueryCustomReturnTypeStep } from "./FunctionQueryCustomReturnTypeStep";
+import { IPostgresFunctionQueryWizardContext } from "./IPostgresFunctionQueryWizardContext";
+
+const customReturnTypePick: QuickPickItem = { label: localize('enterCustomReturnType', '$(pencil) Enter custom return type...') };
+
+export class FunctionQueryReturnTypeStep extends AzureWizardPromptStep<IPostgresFunctionQueryWizardContext> {
+    public async prompt(wizardContext: IPostgresFunctionQueryWizardContext): Promise<void> {
+        const returnTypeQuickPicks: QuickPickItem[] = returnTypes.map(r => { return { label: r }; });
+        returnTypeQuickPicks.push(customReturnTypePick);
+
+        wizardContext.returnTypePick = (await ext.ui.showQuickPick(returnTypeQuickPicks, {
+            placeHolder: localize('selectReturnType', 'Select return type')
+        }));
+    }
+
+    public shouldPrompt(): boolean {
+        return true;
+    }
+
+    public async getSubWizard(wizardContext: IPostgresFunctionQueryWizardContext): Promise<IWizardOptions<IPostgresFunctionQueryWizardContext> | undefined> {
+        if (wizardContext.returnTypePick === customReturnTypePick) {
+            return { promptSteps: [new FunctionQueryCustomReturnTypeStep()] };
+        }
+
+        wizardContext.returnType = wizardContext.returnTypePick.label;
+        return undefined;
+    }
+}
+
+// A subset of return types available on pgAdmin
+const returnTypes: string[] = [
```

Comment:
I paired down the full list of return types. Open to suggestions for showing more/less options
<details>
  <summary>Full list</summary>

```
"char"
"char"[]
abstime
abstime[]
aclitem
aclitem[]
anyarray
anyelement
anyenum
anynonarray
anyrange
bigint
bigint[]
bit
bit varying
bit varying[]
bit[]
boolean
boolean[]
box
box[]
bytea
bytea[]
character
character
character varying
character[]
cid
cid[]
cidr
cidr[]
circle
circle[]
cstring
cstring[]
date
date[]
daterange
daterange[]
double precision
double precision[]
fdw_handler
gtsvector
gtsvector[]
index_am_handler
inet
inet[]
int2vector
int2vector[]
int4range
int4range[]
int8range
int8range[]
integer
integer[]
internal
interval
interval[]
json
json[]
jsonb
jsonb[]
line
line[]
lseg
lseg[]
macaddr
macaddr8
macaddr8[]
macaddr[]
money
money[]
name
name[]
numeric
numeric[]
numrange
numrange[]
oid
oid[]
oidvector
oidvector[]
opaque
path
path[]
pg_aggregate
pg_am
pg_amop
pg_amproc
pg_attrdef
pg_attribute
pg_auth_members
pg_authid
pg_available_extension_versions
pg_available_extensions
pg_buffercache
pg_cast
pg_class
pg_collation
pg_config
pg_constraint
pg_conversion
pg_cursors
pg_database
pg_db_role_setting
pg_ddl_command
pg_default_acl
pg_depend
pg_dependencies
pg_description
pg_enum
pg_event_trigger
pg_extension
pg_file_settings
pg_foreign_data_wrapper
pg_foreign_server
pg_foreign_table
pg_group
pg_hba_file_rules
pg_index
pg_indexes
pg_inherits
pg_init_privs
pg_language
pg_largeobject
pg_largeobject_metadata
pg_locks
pg_lsn
pg_lsn[]
pg_matviews
pg_namespace
pg_ndistinct
pg_node_tree
pg_opclass
pg_operator
pg_opfamily
pg_partitioned_table
pg_pltemplate
pg_policies
pg_policy
pg_prepared_statements
pg_prepared_xacts
pg_proc
pg_publication
pg_publication_rel
pg_publication_tables
pg_range
pg_replication_origin
pg_replication_origin_status
pg_replication_slots
pg_rewrite
pg_roles
pg_rules
pg_seclabel
pg_seclabels
pg_sequence
pg_sequences
pg_settings
pg_shadow
pg_shdepend
pg_shdescription
pg_shseclabel
pg_stat_activity
pg_stat_all_indexes
pg_stat_all_tables
pg_stat_archiver
pg_stat_bgwriter
pg_stat_database
pg_stat_database_conflicts
pg_stat_progress_vacuum
pg_stat_replication
pg_stat_ssl
pg_stat_statements
pg_stat_subscription
pg_stat_sys_indexes
pg_stat_sys_tables
pg_stat_user_functions
pg_stat_user_indexes
pg_stat_user_tables
pg_stat_wal_receiver
pg_stat_xact_all_tables
pg_stat_xact_sys_tables
pg_stat_xact_user_functions
pg_stat_xact_user_tables
pg_statio_all_indexes
pg_statio_all_sequences
pg_statio_all_tables
pg_statio_sys_indexes
pg_statio_sys_sequences
pg_statio_sys_tables
pg_statio_user_indexes
pg_statio_user_sequences
pg_statio_user_tables
pg_statistic
pg_statistic_ext
pg_stats
pg_subscription
pg_subscription_rel
pg_tables
pg_tablespace
pg_timezone_abbrevs
pg_timezone_names
pg_transform
pg_trigger
pg_ts_config
pg_ts_config_map
pg_ts_dict
pg_ts_parser
pg_ts_template
pg_type
pg_user
pg_user_mapping
pg_user_mappings
pg_views
point
point[]
polygon
polygon[]
real
real[]
record
record[]
refcursor
refcursor[]
regclass
regclass[]
regconfig
regconfig[]
regdictionary
regdictionary[]
regnamespace
regnamespace[]
regoper
regoper[]
regoperator
regoperator[]
regproc
regproc[]
regprocedure
regprocedure[]
regrole
regrole[]
regtype
regtype[]
reltime
reltime[]
smallint
smallint[]
smgr
tab
tab[]
tabl
tabl[]
test.tab
text
text[]
tid
tid[]
time with time zone
time with time zone[]
time without time zone
time without time zone[]
timestamp with time zone
timestamp with time zone[]
timestamp without time zone
timestamp without time zone[]
tinterval
tinterval[]
tsm_handler
tsquery
tsquery[]
tsrange
tsrange[]
tstzrange
tstzrange[]
tsvector
tsvector[]
txid_snapshot
txid_snapshot[]
uuid
uuid[]
varying[]</li>
void
xid
xid[]
xml
xml[]
```
</details>